tools_for_todoist/models/event.py

```diff
@@ -117,22 +122,50 @@ def _last_occurrence(self):
 
         if instances.count() == 0:
             return None
-        last_occurence = instances[-1]
+        last_occurrence = instances[-1]
         if not is_allday(start):
-            last_occurence = last_occurence.astimezone(start.tzinfo)
-        return last_occurence.date() if is_allday(start) else last_occurence
+            last_occurrence = last_occurrence.astimezone(start.tzinfo)
+        return last_occurrence.date() if is_allday(start) else last_occurrence
+    
+    def _find_first_non_exception_instance(self, rrule_instances, exception_original_starts):
+        first_moment = datetime.now() if is_allday(self._get_start()) else datetime.now(UTC)
+        inc = True
+        while True:
+            next_regular_occurrence = rrule_instances.after(first_moment, inc=inc)
```

Comment:
use `xafter` and make sure you supported ending event sequences (at some point we should return None if the event is before now or if it ends and all instances are cancelled.